.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: Build
+on:
+  push:
+    branches: [main]
+  pull_request:
+permissions:
+  contents: write
```

Comment:
```suggestion
```
This is not necessary as we set the `permissions` in the `release` workflow, right?